engine/access/rpc/backend/backend.go

```diff
@@ -278,27 +278,31 @@ func executionNodesForBlockID(
 			break
 		}
 
-		select {
-		case <-ctx.Done():
-			return flow.IdentityList{}, err
-		case <-time.After(100 * time.Millisecond << time.Duration(attempt)):
-			//retry after an exponential backoff
-		}
-
 		// if one or less execution receipts may have been received then re-query
 		// in the hope that more might have been received by now
+
+		// log the attempt
 		log.Debug().Int("attempt", attempt).Int("max_attempt", maxAttemptsForExecutionReceipt).
 			Int("execution_receipts_found", len(executorIDs)).
 			Str("block_id", blockID.String()).
 			Msg("insufficient execution receipts")
+
+		select {
+		case <-ctx.Done():
+			return flow.IdentityList{}, err
+		case <-time.After(100 * time.Millisecond << time.Duration(attempt)):
+			//retry after an exponential backoff
+		}
 	}
 
 	if attempt == maxAttemptsForExecutionReceipt {
-		return flow.IdentityList{},
-			fmt.Errorf("number of execution receipts found (%d) less than expected (%d) for block ID %v",
-				len(executorIDs),
-				minExecutionNodesCnt,
-				blockID)
+		log.Info().Int("attempt", attempt).Msg("not enough execution receipt found")
+		// TODO: should we just throw an error instead?
```

Comment:
If we don't have any matching receipts by this point, won't the most likely scenario be that we just don't know who to ask? feels like it might be better to error here.